transport/http/peer.go

```diff
@@ -73,8 +80,32 @@ func (p *httpPeer) isAvailable() bool {
 	return false
 }
 
+func (p *httpPeer) OnSuspect() {
+	now := time.Now().UnixNano()
+	innocentUntil := p.innocentUntilUnixNano.Load()
+
+	// Do not check for connectivity after every request timeout.
+	// Spread them out so they only occur once in every innocence window.
+	if now < innocentUntil {
+		return
```

Comment:
no coverage here

Reply:
Addressed

Reply:
still red according to the coverage, but it could be stale?

Reply:
I added a runtime.Gosched() to the innocence window test and it gets coverage half the time now. I think that ought to be enough, unless there’s something not-too-invasive I can do to force the test and OnSuspect() call to synchronize. For now, I’m going to land this.